src/k8s/pkg/k8sd/setup/util_kubeconfig.go

```diff
@@ -1,40 +1,55 @@
 package setup
 
 import (
-	"encoding/base64"
 	"fmt"
-	"io"
-	"os"
-)
+	"strings"
 
-var (
-	kubeconfigTemplate = mustTemplate("kubeconfig")
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 )
 
-type kubeconfigTemplateConfig struct {
-	CA    string
-	URL   string
-	Token string
-}
+// createConfig generates a Config suitable for our k8s environment.
+func createConfig(token string, server string, caPEM string) *clientcmdapi.Config {
+	config := clientcmdapi.NewConfig()
+
+	// Default to https:// prefix if no http-like scheme is present.
+	// Note: scheme-less host:port isn't a valid url, so no url.Parse here.
+	if !strings.HasPrefix(server, "http") {
+		server = fmt.Sprintf("https://%s", server)
+	}
 
-// renderKubeconfig writes a kubeconfig to the specified writer.
-func renderKubeconfig(writer io.Writer, token string, url string, caPEM string) error {
-	if err := kubeconfigTemplate.Execute(writer, kubeconfigTemplateConfig{
-		CA:    base64.StdEncoding.EncodeToString([]byte(caPEM)),
-		URL:   url,
+	config.Clusters["k8s"] = &clientcmdapi.Cluster{
+		CertificateAuthorityData: []byte(caPEM),
+		Server:                   server,
+	}
+	config.AuthInfos["k8s-user"] = &clientcmdapi.AuthInfo{
 		Token: token,
-	}); err != nil {
-		return fmt.Errorf("failed to render template: %w", err)
 	}
-	return nil
+	config.Contexts["k8s"] = &clientcmdapi.Context{
+		Cluster:  "k8s",
+		AuthInfo: "k8s-user",
+	}
+	config.CurrentContext = "k8s"
+
+	return config
 }
 
 // Kubeconfig writes a kubeconfig file to disk.
 func Kubeconfig(path string, token string, url string, caPEM string) error {
-	file, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	config := createConfig(token, url, caPEM)
+	err := clientcmd.WriteToFile(*config, path)
+	if err != nil {
```

Comment:
```suggestion
	if err := clientcmd.WriteToFile(*config, path); err != nil {
```

Reply:
Also, please keep using `clientcmd.Write()` as we have it already, so that we ensure the right permissions (0600) on the file.

Reply:
> please keep using clientcmd.Write() as we have it already, so that we ensure the right permissions (0600)

@neoaggelos i'd rather let [`clientcmd.WriteToFile` manage the file perms](https://github.com/kubernetes/client-go/blob/v0.29.2/tools/clientcmd/loader.go#L439-L441). today, we're both `0600`, but if a new best practice comes out tomorrow to go `0640`, i think client-go will notice before we do. bonus points that it builds the ppath for us as well.  wdyt?

Reply:
sure, that works